app/src/main/java/io/gnosis/safe/ui/base/BaseStateViewModel.kt

```diff
@@ -45,14 +32,12 @@ abstract class BaseStateViewModel<T : BaseStateViewModel.State>(
             val currentState = currentState()
             val nextState = currentState.run(update)
             // Reset view action if the same
-            if (!forceViewAction && nextState.viewAction === currentState.viewAction) nextState.viewAction = null
-            stateChannel.send(nextState)
+            takeUnless { !forceViewAction && nextState === currentState }?.run {
+                stateChannel.send(nextState)
+            }
         } catch (e: Exception) {
-            // Could not submit update
             Timber.e(e)
         }
     }
 
-    protected fun safeLaunch(errorHandler: CoroutineExceptionHandler = coroutineErrorHandler, block: suspend CoroutineScope.() -> Unit) =
```

Comment:
If you remove this then you can also remove `coroutineErrorHandler` ... I would strongly recomment to have something like this though as it makes error handling a log easier (but also depends what the plan for error handling is ;) )

Reply:
I saw your comments after merging, sorry for that. I will restore this.